tests/test_builtin_traits.py

```diff
@@ -258,6 +262,57 @@ def test_no_terminator_op_with_is_terminator_op():
     op0.verify()
 
 
+@irdl_op_definition
+class IsSingleBlockImplicitTerminatorOp(IRDLOperation):
+    """
+    An operation that implements terminator to be used with an operation that uses the
+    SingleBlockImplicitTerminator trait.
+    """
+
+    name = "test.is_single_block_implicit_terminator"
+
+    # TODO fix circular reference
+    # traits = frozenset([HasParent(HasSingleBlockImplicitTerminatorOp), IsTerminator()])
```

Comment:
Not sure what is the best way to resolve this circular dependency.

Reply:
Let's leave it for now, I have an idea on how to make that work, but that will require significant changes in the `irdl.py` file :/

Reply:
OK, no problem, I opened an issue to track this https://github.com/xdslproject/xdsl/issues/1218